osu.Game/Rulesets/Edit/ScrollingHitObjectComposer.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.UI.Scrolling;
+using osu.Game.Screens.Edit.Components.TernaryButtons;
+using osuTK;
+
+namespace osu.Game.Rulesets.Edit
+{
+    public abstract partial class ScrollingHitObjectComposer<TObject> : HitObjectComposer<TObject>
```

Comment:
This though I'm feeling kind of bearish about since it's starting to feel like a potential future inheritance finger trap. But I trust that we'll be able to recognise when it gets bad enough so probably fine for now...

Reply:
yeah.